libcontainer/cgroups/fs/memory.go

```diff
@@ -96,9 +96,18 @@ func setKernelMemory(path string, kernelMemoryLimit int64) error {
 }
 
 func setMemoryAndSwap(path string, cgroup *configs.Cgroup) error {
-	// When memory and swap memory are both set, we need to handle the cases
-	// for updating container.
-	if cgroup.Resources.Memory != 0 && cgroup.Resources.MemorySwap > 0 {
+	// Memeory set to '-1' which is unlimited. In this situation we need to update swap to -1
+	// to avoid kernel validation failures.
+	if cgroup.Resources.Memory == -1 {
+		if err := writeFile(path, "memory.memsw.limit_in_bytes", strconv.FormatInt(-1, 10)); err != nil {
```

Comment:
It's not "unnecessary overhead". `libcontainer` is *meant* to be stupid (though clever enough to know when to be clever about it), specifically this means that we in general don't take initiative past what the user has told us to do -- it's not `libcontainer`'s job to change the settings that a user asked for.

If you wanted to idiot-proof this then you could put it inside `libcontainer/specconv` so that runC makes this easier for users. But then again, I'm not even sure if we should be doing *that* to be honest.